datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -314,8 +314,13 @@ fn coerced_from<'a>(
     // match Dictionary first
     match (type_into, type_from) {
         // coerced dictionary first
-        (cur_type, Dictionary(_, value_type)) | (Dictionary(_, value_type), cur_type)
-            if coerced_from(cur_type, value_type).is_some() =>
```

Comment:
When coercing into dictionary type, the `type_into` and `type_from` parameters are in incorrect order.